directanswercards/documentsearch-standard/component.js

```diff
@@ -54,6 +59,27 @@ class documentsearch_standardComponent extends BaseDirectAnswerCard['documentsea
   static defaultTemplateName (config) {
     return 'directanswercards/documentsearch-standard';
   }
+
+  /**
+   * construct a URL that links to a specific portion of a page, using a text snippet provided in the URL.
+   * @param {Object} snippet the snippet for the document search direct answer
+   * @param {string} baseUrl website or landingPageURL from the entity related to the snippet
+   * @returns a URL with text fragment URI component attached
+   */
+  getUrlWithTextHighlight(snippet, baseUrl) {
+    //Find the surrounding sentence of the snippet
+    let sentenceStart = snippet.matchedSubstrings[0].offset;
+    let sentenceEnd = sentenceStart + snippet.matchedSubstrings[0].length;
+    while (!/[.\n!?]/.test(snippet.value[sentenceStart]) && sentenceStart > 0) {
```

Comment:
could we have maybe a variable name for the regex or just some comments saying what it's doing

Reply:
will do!